### PR TITLE
Add comprehensive sports performance tools and enhanced formatting

### PR DESCRIPTION
**Pull request by scorphus**
# Add comprehensive sports performance tools and enhanced formatting

## Summary

This PR significantly expands the intervals-mcp-server with new MCP tools focused on sports performance analysis and enhanced data formatting. The changes provide comprehensive access to Intervals.icu's training data with clean, agent-friendly output formats.

### 🕐 Date & Time Tools

- **get_current_date_and_time_info**: Enhanced date/time tool with timezone support
- **calculate_date_info**: Calculate date information for any given date
- Comprehensive timezone information including UTC conversion

### 🏃‍♂️ Sports Performance Tools

- **get_races**: Retrieve race events for athletes
- **get_power_curves**: Get power curve data for cycling performance analysis
- **get_activity_power_curves**: Activity-specific power curve analysis
- **get_pace_curves**: Pace curve data for running performance
- **get_activity_pace_curve**: Activity-specific pace analysis
- **get_power_hr_curve**: Power vs heart rate correlation analysis
- **get_activity_power_vs_hr**: Activity power/HR relationship
- **get_activity_hr_curve**: Heart rate curve analysis

### 💪 Workout Management

- **create_workout**: Create structured workouts with Intervals.icu format
- **get_workout_format_examples**: Access to workout format documentation
- Support for power zones, pace zones, time/distance intervals

### 📊 Enhanced Data Formatting

- **Athlete Profile Formatting**: Transform raw athlete JSON into structured Markdown
- **Sport-specific sections**: Cycling, Running, Swimming with training zones
- **Training thresholds**: FTP, LTHR, Max HR, threshold pace display
- **Zone breakdowns**: Power zones, HR zones, pace zones with clear ranges
- Clean, agent-optimized output without emojis

### 🧪 Comprehensive Testing

- Added test coverage for most new tools
- Formatting function validation
- Error handling and edge case coverage

### 🧹 Code Quality

- Moved all imports to top-level (no more inline imports)
- Black and isort formatting applied
- Enhanced error handling with specific HTTP status messages
- Improved type hints and documentation

### 🔧 Infrastructure

- Added development dependencies (black, isort)
- Enhanced pyproject.toml configuration
- Comprehensive workout samples documentation

## API Coverage Expansion

This PR dramatically expands the server's coverage of the Intervals.icu API, adding tools for:

- Training load and performance curves
- Workout creation and management
- Advanced athlete data analysis
- Date/time utilities for training planning
- Race and event management

## Breaking Changes

- `get_athlete` now returns formatted Markdown instead of raw JSON
- `get_current_date_info` renamed to `get_current_date_and_time_info`

## Test Plan

- [x] All existing tests pass
- [x] Basic tests added for new formatting functions
- [x] Input validation for null/empty/invalid inputs
- [x] Error handling for invalid date formats

## Yet to be done

- [ ] Add remaining test for uncovered tools
- [ ] Document the use of Black and isort; if they're going to be accepted as part of this CL
- [ ] Refactor other tools to return tailored Markdown content instead of raw JSON

## Notes

This is a draft PR to gather feedback on the direction and implementation. The changes significantly enhance the server's capabilities for sports performance analysis while maintaining clean, agent-friendly interfaces.

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

**Comment by scorphus** [comment]
I'm happy to contribute! Thank you for this MCP, really! It's great and has impacted my training in such a positive way! It's great that you used intertools! I'm glad it was of use :-)

First question I have: Are you okay with the introduction of Black and isort? There are still a couple of things to be done in that regard.

Also, please LMK if this CL is too big. I can prolly break it into smaller, incremental PRs.

Thanks!

File: src/intervals_mcp_server/utils/formatting.py [hunk 1]
```diff
@@ -356,3 +356,170 @@ def format_intervals(intervals_data: dict[str, Any]) -> str:
...
+    Returns:
+        A formatted Markdown string representation of the athlete data
+    """
+    if not athlete or not isinstance(athlete, dict):
+        return "No athlete data available"
+
+    # Basic athlete information
+    weight_str = (
+        f"{athlete.get('icu_weight', athlete.get('weight', 'N/A'))} kg"
+        if athlete.get("icu_weight") or athlete.get("weight")
```
[the remark this answers is not in the document]
Comment (scorphus) [reply src/intervals_mcp_server/utils/formatting.py:376]: Suggestion not taken. Copilot is being technically correct about Python truthiness (0 is falsy), but it's missing the domain context that 0 weight is meaningless for an athlete. The current code is actually doing the right thing by treating 0 weight as "no valid weight data."

**Comment by scorphus** [comment]
Hey, @mvilanova. I'm afraid the changes here might have gone silent or flown under the radar enough that you missed them. I also appreciate the fact that you might be busy with life and all the cycling in such nice weather lately (hopefully). Either way, I hope you don't mind the ping here ☺️

**Comment by scorphus** [comment]
> Check out my fork here that has all current PRs merged plus some other improvements: https://github.com/spraot/intervals-mcp-server

Amazing! Let's get that one going! Next up are improvements to the workout creation/recommendation, since I reckon the one I wrote works better!
